### ISSUE TITLE
--filter <package_name>... does not recognize workspace dependencies that don't have a "version"

### ISSUE THREAD

**Issue by jacques-ssense** [pnpm/pnpm#4123]
### Code to reproduce the issue:

Given the following 2 packages in a mono-repo using pnpm Workspaces:

first package.json:

```json
{
  "name": "package-A",
  "dependencies": {
    "package-B": "workspace:*"
  }
}
```

second package.json:

```json
{
  "name": "package-B"
}
```

### Expected behavior:

```bash
$ pnpm --filter package-A... exec grep name package.json
  "name": "package-A"
  "name": "package-B"
```

### Actual behavior:

```bash
$ pnpm --filter package-A... exec grep name package.json
  "name": "package-A"
```

### Additional information:

If I add a "version" to package B, then it works as expected.

 - `node -v` prints: v14.17.4
 - Windows, macOS, or Linux?: Ubuntu Linux

**Comment by jacques-ssense** [comment]
Apologies if this is not really a bug. The packages I am working on are not intended to be published to an NPM repository, so I thought that I could leave out the version.

Or maybe another option would be to mention that "version" is a required field for this type of filter to work.

Oh, I just looked up the package.json doc to see if version is always mandatory, and it says that it is optional if you don't intend to publish: https://docs.npmjs.com/cli/v6/configuring-npm/package-json#version

**Comment by zkochan** [comment]
This might be fixed by https://github.com/pnpm/pnpm/issues/3933

try pnpm v6.24.0-1

**Comment by fireairforce** [comment]
Fixed at v6.24.0, works fine now: https://github.com/pnpm/pnpm/releases/tag/v6.24.0

**Comment by jacques-ssense** [comment]
ok, great! thank you!
